mmv1/products/discoveryengine/SearchEngine.yaml

```
@@ -0,0 +1,153 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'SearchEngine'
+description: |
+  Vertex AI Search and Conversation can be used to create a search engine or a chat application by connecting it with a datastore
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Create a Search Engine': 'https://cloud.google.com/generative-ai-app-builder/docs/create-engine-es'
+  api: 'https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.collections.engines'
+
+base_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+self_link: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines?engineId={{engine_id}}'
+delete_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+
+update_verb: :PATCH
+update_mask: true
+
+autogen_async: true
+
+async: !ruby/object:Api::OpAsync
+  actions: ['create', 'delete']
+  operation: !ruby/object:Api::OpAsync::Operation
+    base_url: '{{op_id}}'
+  result: !ruby/object:Api::OpAsync::Result
+    resource_inside_response: true
+
+import_format:
+  [
+    'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}',
+  ]
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  encoder: templates/terraform/encoders/discovery_engine_search_engine_hardcode_solution_type.go.erb
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "discoveryengine_searchengine_basic"
+    primary_resource_id: 'basic'
+    primary_resource_name:
+      'fmt.Sprintf("tf_test_search_engine%s", context["random_suffix"])'
+    vars:
+      engine_id: "example-engine-id"
+      data_store_id: "example-datastore-id"
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'engine_id'
```

```suggestion
    name: 'engineId'
```
Sorry for missing it. Should not affect generated code downstream though